### ISSUE TITLE
MNT: Have a plan to retire vo_conesearch

### ISSUE THREAD

**Issue by pllim** [astropy/astroquery#2027]
At Astropy Coordination Meeting 2021, @tomdonaldson mentioned a desire to replace `vo_conesearch` with something else (probably with something in `pyvo`?). We need to formulate a transition plan with a timeline.

**Comment by bsipocz** [comment]
Would you have that something else in astroquery or rather a "how-to" for using pyvo for doing cone search?

**Comment by tomdonaldson** [comment]
Since `pyvo` already supports doing a VO standard cone search, this may mostly be an exercise in clearly documenting the alternative approach in the `vo_conesearch` documentation, then deprecating `vo_conesearch`.  I want to look more carefully at what value-added features `vo_conesearch` offers to see if there are things that should be added to `pyvo`.

Based on the decisions we've had before regarding VO image searches and TAP searches, I think it is most consistent for the generic cone search capability to live in pyvo.  Then, as with TAP, if an astroquery API for a specific resource wants to utilize that feature, it can.

**Comment by pllim** [comment]
In the spirit of [PEP 20](https://www.python.org/dev/peps/pep-0020/), "There should be one-- and preferably only one --obvious way to do it," I agree with you that PyVO should be the true home of cone search, and will be happy to work with you to make that happen. We can start with documentation, but I still want a path towards deprecation and eventual removal.

Besides the API here, I do keep a daily cron job to vet select services but I am really not sure if anyone is even using it.

TBH, `vo_conesearch` is at the "if it ain't broken, don't fix it" stage, as I am not really getting paid to maintain it anymore.

**Comment by bsipocz** [comment]
@tomdonaldson and @pllim - Would be nice to deprecate the module in favour of pyvo in the next release (coming out in the next few months or so), so we can actually remove it sometimes in the future. What do you think? Is there a plug-and-play tutorial or similar that can be linked-to in the docs as an alternative?

**Comment by pllim** [comment]
I will be happy to help but pyvo devs need to take the lead on this. I have not been involved in VO stuff for years. I just pop in once in a while to put out fires. 😬

**Comment by bsipocz** [comment]
I'll cross open an issue in pyvo, too.

**Comment by pllim** [comment]
Looks like most services are "dead" on https://astroconda.org/aux/vo_databases/ . Only `gsss.stsci.edu` is listed now. I don't have time to maintain this, so hopefully no one is using this anymore? 😬

**Comment by bsipocz** [comment]
Ok, let's gut it out.

**Comment by pllim** [comment]
@bsipocz what is the plan now? Please advise. Thanks!

**Comment by bsipocz** [comment]
If you open up a removal pr I would merge it. If it has to wait for me to do it, it may take a bit longer as it's not at the top of my own list to do.

**Comment by bsipocz** [comment]
Or we could start with a deprecation for now and full removal later in the autumn. I suppose that would be a bit more preferable if there is still some working components around.

**Comment by pllim** [comment]
There is a replacement in pyvo, right?

**Comment by bsipocz** [comment]
People can do SCS with pyvo yes. It's not a 1-to-1 direct replacement though, at least in my understanding (but to be honest, I never saw what this module was being used for, or if it was used at all)

**Comment by pllim** [comment]
I think it is probably more responsible to deprecate first. I can put this on my queue but it is not high priority for me either. 🤷‍♀️ 

I also asked in https://groups.google.com/g/astropy-dev/c/uiMgYNYcdd8 , just in case.

**Comment by bsipocz** [comment]
If we go with deprecation first, then we don't really need to wait for any feedback on those threads; if there is a loud roar, in the worse case we can roll back the deprecation.

**Comment by tomdonaldson** [comment]
I agree with the deprecation approach.  In the unlikely event anyone is using it, they will have a chance to transition to another approach, and we could certainly help with any specific use cases that some up.

**Comment by jehturner** [comment]
I believe we are still using this (just checking with colleagues), so please do deprecate it first; I'm not sure whether DRAGONS 4.2 or AstroPy 7.3/8.0 will come out first.

**Comment by bsipocz** [comment]
@jehturner - just a reminder, this has nothing to do with the astropy version.

**Comment by jehturner** [comment]
Sorry, quite right! Don't know about the astroquery release schedule, but if it's deprecated first, I think we should be able to manage.

**Comment by bsipocz** [comment]
OK, thanks for the feedback it's super helpful.

Then saw we are deprecating it first then and then I expect an at least a 6month long deprecation period (as the release schedule is not as predetermined here as for astropy as well as we're planning to change versioning themes, too).

**Comment by pllim** [comment]
So before anyone panics, the conesearch algorithm still works, but the server-side daily vetting is dying. So as long as you use your own list of services by providing custom JSON file, you should be okay for now.

**Comment by jehturner** [comment]
Does the `pyvo` version not accept results from the server in the same format(s) as astroquery? If not, things might be more complicated on our end (have to check), as the back ends include our own catalogue servers, as well as VizieR. But the docs for both describe a "simple cone search API", which sounds the same.

**Comment by bsipocz** [comment]
I am not aware the results of any direct comparisons between the two. I suspect they are not 1-on-1 replacements though as checking on this was floated a few times but even as it got on individual todo lists it always had a rather low priority.

**Comment by pllim** [comment]
Well, if people actually still using it, maybe we can keep it while longer but I can slap a warning on using the default  JSON file.

**Comment by bsipocz** [comment]
I would really like it gone and nudge people towards using pyvo directly. This is legacy code and I really prefer not to do any maintenance on it but focus any efforts on community tools like pyvo.

**Comment by bsipocz** [comment]
(and this one is really an odd ball in astroquery and only ended up here as a refugee after being kicked out from astropy. It actually should have not ended up here at all)

**Comment by pllim** [comment]
A transition guide might be nice. 🤔

**Comment by bsipocz** [comment]
> A transition guide might be nice. 🤔

Who is writing it?  I was floated as an idea before, but always stuck at at that point, and honestly at this point I may not want that existence/non-existence of that guide to be the blocker on ultimately removing this from astroquery.

**Comment by pllim** [comment]
> Who is writing it?

I can assist from `vo_conesearch` side but will need to work with someone familiar with the cone search over at `pyvo`. Probably a hack day sort of thing.

**Comment by ManonMarchand** [comment]
`vo_conesearch` has server-side functionalities that don't exist in pyVO. We can document how people using the client-side can switch, but the rest won't have replacements (which is ok if no one uses it). I can help write/review such a document.

**Comment by pllim** [comment]
The server validation stuff was nice in principle when it was envisioned but I think in practice, people already have a list of servers they trust anyway, so my guess is no one actually uses it. @jehturner , does your team use it?

**Comment by pllim** [comment]
@jehturner , in regards to https://github.com/astropy/astroquery/issues/2027#issuecomment-3928938483 , do you have a list of requirements or desired features, so we can be sure this transition will not be too hard on your team? Thanks!

**Comment by jehturner** [comment]
I don't think we're using any special validation; our code queries specific URLs anyway. We really just do a simple `conesearch()` in one place in the code and we've now replaced that with `pyvo.dal.SCSService(url).search()` in the DRAGONS repo, so hopefully it's just a case of doing enough testing 🤞. Thanks!

**Comment by pllim** [comment]
OK I have not heard any objection, so I can take a stab at this.

**Comment by pllim** [comment]
@bsipocz , do you want something like https://github.com/astropy/astropy/pull/5904 , or do you just want the warning to appear on import from like [vo_conesearch/\_\_init\_\_.py](https://github.com/astropy/astroquery/blob/main/astroquery/vo_conesearch/__init__.py) ?

**Comment by bsipocz** [comment]
I would leave it to your judgement, either would work.
(I prefer the import time, the option in astropy is just unusefully verbose, all we want just one warning at the very top of the usage. )

**Comment by pllim** [comment]
Please point stakeholders to review https://github.com/astropy/astroquery/pull/3548 . Thanks!
